### PR TITLE
[SPARK-41443][SQL] Assign a name to the error class _LEGACY_ERROR_TEMP_1061

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
In the PR, I propose to assign the name COLUMN_NOT_FOUND to the error class _LEGACY_ERROR_TEMP_1061.

### Why are the changes needed?
Proper names of error classes should improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Add new UT.
Pass GA.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -109,6 +109,11 @@
       "The column <columnName> already exists. Consider to choose another name or rename the existing column."
     ]
   },
+  "COLUMN_NOT_EXISTS" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:112]: How about `COLUMN_NOT_FOUND`. This is more consistent to other error classes:
```
$ grep NOT_FOUND ./core/src/main/resources/error/error-classes.json
  "FIELD_NOT_FOUND" : {
  "INDEX_NOT_FOUND" : {
  "PARTITIONS_NOT_FOUND" : {
  "PATH_NOT_FOUND" : {
  "PROTOBUF_DEPENDENCY_NOT_FOUND" : {
  "PROTOBUF_DESCRIPTOR_FILE_NOT_FOUND" : {
  "PROTOBUF_MESSAGE_NOT_FOUND" : {
  "RENAME_SRC_PATH_NOT_FOUND" : {
  "ROUTINE_NOT_FOUND" : {
  "SCHEMA_NOT_FOUND" : {
  "TABLE_OR_VIEW_NOT_FOUND" : {
      "CORRELATED_COLUMN_NOT_FOUND" : {
  "VIEW_NOT_FOUND" : {
```

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala [hunk 2]
```diff
@@ -155,7 +155,7 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
     case DescribeColumn(ResolvedV1TableIdentifier(ident), column, isExtended, output) =>
       column match {
         case u: UnresolvedAttribute =>
-          throw QueryCompilationErrors.columnDoesNotExistError(u.name)
+          throw QueryCompilationErrors.columnNotExistError(u.name)
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala:158]: Should be `NotExists` but let's rename it to `columnNotFoundError`

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -109,6 +109,11 @@
       "The column <columnName> already exists. Consider to choose another name or rename the existing column."
     ]
   },
+  "COLUMN_NOT_FOUND" : {
+    "message" : [
+      "The column <colName> does not exist."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:114]:
```suggestion
      "The column <colName> cannot be found. Verify the spelling and correctness of the column name according to the SQL config \"spark.sql.caseSensitive\"."
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 3 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: Done

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala (hunk 2 above)
Comment (panbingkun) [reply sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala]: Done

**Comment by MaxGekk** [comment]
@panbingkun Could you fix the test failure, seems it is related to your changes:
```
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: "[COLUMN_NOT_FOUND] The column `some_random_column` cannot be found. Verify the spelling and correctness of the column name according to the SQL config "spark.sql.caseSensitive"." did not contain "does not exist"
	at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
	at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
	at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
	at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
	at org.apache.spark.sql.StatisticsCollectionSuite.$anonfun$new$15(StatisticsCollectionSuite.scala:136)
```

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by panbingkun** [comment]
> @panbingkun Could you fix the test failure, seems it is related to your changes:
> 
> ```
> sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: "[COLUMN_NOT_FOUND] The column `some_random_column` cannot be found. Verify the spelling and correctness of the column name according to the SQL config "spark.sql.caseSensitive"." did not contain "does not exist"
> 	at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
> 	at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
> 	at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
> 	at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
> 	at org.apache.spark.sql.StatisticsCollectionSuite.$anonfun$new$15(StatisticsCollectionSuite.scala:136)
> ```

Done, Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**